core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -68,104 +73,174 @@ open class Amplitude internal constructor(
     }
 
     @Deprecated("Please use 'track' instead.", ReplaceWith("track"))
-    fun logEvent(event: BaseEvent) {
-        track(event)
+    fun logEvent(event: BaseEvent): Amplitude {
+        return track(event)
     }
 
-    fun track(event: BaseEvent, callback: EventCallBack? = null) {
+    /**
+     * Track an event.
+     *
+     * @param event the event
+     * @param callback the optional event callback
+     * @return the Amplitude
+     */
+    fun track(event: BaseEvent, callback: EventCallBack? = null): Amplitude {
         callback ?. let {
             event.callback = it
         }
         process(event)
+        return this
     }
 
-    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null) {
+    /**
+     * Log event with the specified event type, event properties, and optional event options.
+     *
+     * @param eventType the event type
+     * @param eventProperties the event properties
+     * @param options optional event options
+     * @return the Amplitude
+     */
+    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null): Amplitude {
         val event = BaseEvent()
         event.eventType = eventType
         event.eventProperties = eventProperties
         options ?. let {
             event.mergeEventOptions(it)
         }
         process(event)
+        return this
     }
 
     /**
-     * Identify. Use this to send an Identify object containing user property operations to Amplitude server.
+     * Identify lets you to send an Identify object containing user property operations to Amplitude server.
+     * You can modify user properties by calling this api.
+     *
+     * @param identify identify object
+     * @param options optional event options
+     * @return the Amplitude
      */
-    fun identify(identify: Identify, options: EventOptions? = null) {
+    fun identify(identify: Identify, options: EventOptions? = null): Amplitude {
         val event = IdentifyEvent()
         event.userProperties = identify.properties
         options ?. let {
             event.mergeEventOptions(it)
         }
         process(event)
+        return this
     }
 
-    fun identify(userId: String) {
+    /**
+     * Set the user id (can be null).
+     *
+     * @param userId custom user id
+     * @return the Amplitude
+     */
+    fun setUserId(userId: String?): Amplitude {
         this.idContainer.identityManager.editIdentity().setUserId(userId).commit()
+        return this
     }
 
-    fun setDeviceId(deviceId: String) {
+    /**
+     * Sets a custom device id. <b>Note: only do this if you know what you are doing!</b>
+     *
+     * @param deviceId custom device id
+     * @return the Amplitude
```

Comment:
the Amplitude instance?
Same for others

Reply:
Sounds good, already updated